scipy/spatial/ckdtree/src/query.cxx

```diff
@@ -1,10 +1,5 @@
-/*
- * This would break SciPy with NumPy 1.6 so just accept the compiler
- * warning for now.
- * #define NPY_NO_DEPRECATED_API NPY_1_9_API_VERSION 
- *
- */
- 
+#define NPY_NO_DEPRECATED_API NPY_1_9_API_VERSION
```

Comment:
No, that looks wrong to me. Should be done in the spatial setup.py with ``numpy_nodepr_api``. Example: https://github.com/scipy/scipy/blob/master/scipy/optimize/setup.py#L20

IIRC the reason was that before numpy 1.10.0 the numpy deprecation macro did something quite unwanted. It's a while ago, but the detail should be in the PR where I added ``numpy_nodepr_api``.